netkan/netkan/common.py

```diff
@@ -1,24 +1,42 @@
-from typing import List, Iterable, Dict, Any, IO
-import boto3  # pylint: disable=unused-import
+import logging
 
+from typing import Dict, List, Iterable, IO, Type, TYPE_CHECKING, Union
+from types import TracebackType
+
+import boto3
 import requests
 import github
 from git import Repo
 
 from .metadata import Netkan
-from .repos import NetkanRepo
+from .repos import CkanMetaRepo, NetkanRepo
+
+if TYPE_CHECKING:
+    from mypy_boto3_sqs.service_resource import Message
+    from mypy_boto3_sqs.type_defs import (
+        DeleteMessageBatchRequestEntryTypeDef,
+        SendMessageBatchRequestEntryTypeDef,
+    )
+    from .cli.common import Game, SharedArgs
+else:
+    Game = object
+    Message = object
+    DeleteMessageBatchRequestEntryTypeDef = object
+    SendMessageBatchRequestEntryTypeDef = object
+    SharedArgs = object
```

Comment:
This just allows us to accurately type check it with dynamic stuff that comes with boto3. This gives us type checking without breaking the runtime code. See -> https://mypy-boto3.readthedocs.io/en/latest/#pylint-compatibility

Reply:
What's the benefit over importing the types unconditionally? I.e., just doing this at the top like normal:

```python
from mypy_boto3_sqs.service_resource import Message
from mypy_boto3_sqs.type_defs import (
    DeleteMessageBatchRequestEntryTypeDef,
    SendMessageBatchRequestEntryTypeDef,
)
from .cli.common import Game, SharedArgs
```

Reply:
Because we don't actually install the types for the deployment, they're entirely unnecessary. It allows us to deploy the code, whilst retaining the ability to type check during test/development. It is a nuance of the way type checking has been introduced to Python, and not an uncommon pattern.